### DISCUSSION TITLE
How enabled sorting without fields object

### DISCUSSION THREAD

**Discussion by Eliophot** [bootstrap-vue/bootstrap-vue#6562] [Q&A] [answered]
Hello,
Is it possible to activate sorting other than in the fields object?
I don't know in advance the number of fields...
Perhaps in the slot : 
```
<template #head()="data">
```
Thanks

**Comment by Hiws** [comment, marked as answer]
I don't believe it's possible without passing in a fields array.

Assuming you're currently utilizing the auto generated fields by only passing in `items`.
You could utilize the same method to generate a fields array which you can then pass to your table.

```js
<b-table :items="items" :fields="fields" />

{
  computed: {
    fields() {
      if(this.items.length === 0) return [];
      const item = this.items[0];
      return Object.keys(item).map(field => ({ key: field, sortable: true }));
    }
  },
  data() {
    return {
      items: []
    }
  }
}
```

The above example will generate a field array based on the first object in your `items` array.
It will then map the keys and add `sortable: true` to all the fields.

This should create fields and enable sorting for all fields automatically based on your data.

**Reply by Eliophot** [reply]
I do that thanks !
